src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/TransactionTypesSection/ContractInteractionSection.tsx

```diff
@@ -74,16 +202,50 @@ const ContractInteractionSection = ({
       </DialogSection>
       <DialogSection appearance={{ theme: 'sidePadding' }}>
         <div className={styles.contractFunctionSelectorContainer}>
-          {/* @TODO: Connect available contract functions data with picker */}
           <Select
             label={MSG.functionLabel}
             name={`transactions.${transactionFormIndex}.contractFunction`}
             appearance={{ theme: 'grey', width: 'fluid' }}
             placeholder={MSG.functionPlaceholder}
             disabled={disabledInput}
+            options={formattedMethodOptions}
+            onChange={(value) => {
+              const updatedSelectedContractMethods = {
+                ...selectedContractMethods,
+                [transactionFormIndex]: usefulMethods.find(
+                  (method) => method.name === value,
+                ),
+              };
+              handleSelectedContractMethods(updatedSelectedContractMethods);
+            }}
           />
         </div>
       </DialogSection>
+      {selectedContractMethods[transactionFormIndex]?.inputs?.map((input) => (
+        <DialogSection
+          key={`${input.name}-${input.type}`}
+          appearance={{ theme: 'sidePadding' }}
+        >
+          <div className={styles.inputParamContainer}>
+            <Input
+              label={`${input.name} (${input.type})`}
+              name={`transactions.${transactionFormIndex}.${input.name}`}
+              appearance={{ colorSchema: 'grey', theme: 'fat' }}
+              disabled={disabledInput}
+              placeholder={`${input.name} (${input.type})`}
+              formattingOptions={
+                input.type === 'uint256' || input.type === 'int256'
```

Comment:
If you changed this so it matched a type containing `uint/int` it would cover all the int/uint permutations. Then there'd only be the max safe value left to test in a separate issue.